app/src/main/kotlin/com/wire/android/ui/common/ClickableText.kt

```diff
@@ -0,0 +1,139 @@
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ *
+ */
+
+package com.wire.android.ui.common
+
+import android.os.Build
+import android.text.SpannableString
+import android.text.style.URLSpan
+import android.text.util.Linkify
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.material3.LocalTextStyle
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.style.TextOverflow
+
+@Composable
+fun ClickableText(
+    text: AnnotatedString,
+    modifier: Modifier = Modifier,
+    color: Color = Color.Unspecified,
+    textDecoration: TextDecoration? = null,
+    textAlign: TextAlign? = null,
+    overflow: TextOverflow = TextOverflow.Clip,
+    softWrap: Boolean = true,
+    maxLines: Int = Int.MAX_VALUE,
+    onTextLayout: (TextLayoutResult) -> Unit = {},
+    style: TextStyle = LocalTextStyle.current,
+    onClick: (Int) -> Unit,
+    onLongClick: (() -> Unit)? = null,
+) {
+    val layoutResult = remember { mutableStateOf<TextLayoutResult?>(null) }
+
+    // to provide current lambdas and states otherwise it will use the ones it got when starting a LaunchedEffect inside pointerInput
+    val currentOnClick by rememberUpdatedState(newValue = onClick)
+    val currentOnLongClick by rememberUpdatedState(newValue = onLongClick)
+    val currentLayoutResult by rememberUpdatedState(layoutResult)
+
+    val pressIndicator = Modifier.pointerInput(Unit) {
+        detectTapGestures(
+            onTap = { pos ->
+                currentLayoutResult.value?.let { layoutResult ->
+                    currentOnClick(layoutResult.getOffsetForPosition(pos))
+                }
+            },
+            onLongPress = { currentOnLongClick?.invoke() }
+        )
+    }
+    Text(
+        text = text,
+        modifier = modifier.then(pressIndicator),
+        color = color,
+        textDecoration = textDecoration,
+        textAlign = textAlign,
+        overflow = overflow,
+        softWrap = softWrap,
+        maxLines = maxLines,
+        onTextLayout = {
+            layoutResult.value = it
+            onTextLayout(it)
+        },
+        style = style
+    )
+}
+
+data class LinkInfo(
+    val url: String,
+    val start: Int,
+    val end: Int
+)
+
+class SpannableStr(source: CharSequence) : SpannableString(source) {
```

Comment:
Don't abbreviate, it's responsible for links, right? So maybe `LinkSpannableString`